src/wp-includes/block-supports/layout.php

```diff
@@ -140,6 +140,40 @@ function wp_get_layout_style( $selector, $layout, $has_block_gap_support = false
 	return $style;
 }
 
+/**
+ * Generates the utility classnames for the given blocks layout attributes.
+ * This method was primarily added to reintroduce classnames that were removed
+ * in the 5.9 release (https://github.com/WordPress/gutenberg/issues/38719), rather
+ * than providing an extensive list of all possible layout classes. The plan is to
+ * have the style engine generate a more extensive list of utility classnames which
+ * will then replace this method.
+ *
+ * @param array $block_attributes Array of block attributes.
+ *
+ * @return array Array of CSS classname strings.
+ */
+function gutenberg_get_layout_classes( $block_attributes ) {
+	$class_names = array();
+
+	if ( empty( $block_attributes['layout'] ) ) {
+		return $class_names;
+	}
+
+	if ( ! empty( $block_attributes['layout']['orientation'] ) ) {
+		$class_names[] = 'is-' . sanitize_title( $block_attributes['layout']['orientation'] );
+	}
+
+	if ( ! empty( $block_attributes['layout']['justifyContent'] ) ) {
+		$class_names[] = 'is-content-justification-' . sanitize_title( $block_attributes['layout']['justifyContent'] );
+	}
+
+	if ( ! empty( $block_attributes['layout']['flexWrap'] ) && 'nowrap' === $block_attributes['layout']['flexWrap'] ) {
+		$class_names[] = 'is-nowrap';
+	}
+
+	return $class_names;
```

Comment:
A filter may be helpful here too.

Reply:
What would the use case be for adding a filter here? If it's effectively a boolean (so that folks can easily opt-out of injecting classnames), I could imagine that being useful for themes that want to switch off the feature.

Something I'd be slightly wary of, is if we were to introduce a filter that folks then use as a sort of API to inject other arbitrary classnames into the output. There's more (experimental) exploration in iterating on how the Layout support works in https://github.com/WordPress/gutenberg/pull/40875, and I'm keen to ensure that we can continue to make changes to the structure and output of the classnames, without having to factor in whether or not the output is being mutated in some way 🤔

On the other hand, adding a filter is a compelling idea, it might wind up being that a good way to design it is to use a filter that receives the `$block_attributes` and then outputs a list of classnames so that plugins can mutate it how they wish, but I wasn't sure if that might lock us in, since the feature is undergoing a fair few exploratory changes.

Reply:
> What would the use case be for adding a filter here?

I was thinking of something along the lines of the [`body_class`](http://developer.wordpress.org/reference/hooks/body_class/) and [`post_class`](http://developer.wordpress.org/reference/functions/post_class/) filters that allow folks to add or remove arbitrary classes (an empty array being the opt-out). So, exactly the thing you are weary of adding for while exploring the layout issues further 😉 

Let's keep the filter on the todo list for now to avoid the potential for accidentally locking an API in before all the implications are considered.

Thanks for catching this.

Reply:
> Let's keep the filter on the todo list for now to avoid the potential for accidentally locking an API in before all the implications are considered.

Excellent, thanks Peter! I've added a link to this thread in the Gutenberg tracking issue for the Layout block support (https://github.com/WordPress/gutenberg/issues/39336) under the "TODO" section, so that we don't forget 🙂